sources/rest_api/__init__.py

```diff
@@ -0,0 +1,768 @@
+                paginator=paginator,
+            )
+
+        else:
+            predecessor = resources[resolved_param.resolve_config.resource_name]
+
+            param_name = resolved_param.param_name
+            request_params.pop(param_name, None)
+
+            def paginate_dependent_resource(
+                items,
+                method,
+                path,
+                params,
+                paginator,
+                param_name=param_name,
+                field_path=resolved_param.resolve_config.field_path,
+            ):
+                items = items or []
+                for item in items:
+                    formatted_path = path.format(**{param_name: item[field_path]})
+                    parent_resource_name = resolved_param.resolve_config.resource_name
+
+                    parent_record = (
+                        {
+                            make_parent_key_name(parent_resource_name, key): item[key]
+                            for key in include_from_parent
+                        }
+                        if include_from_parent
+                        else None
+                    )
+
+                    for child_page in client.paginate(
+                        method=method,
+                        path=formatted_path,
+                        params=params,
+                        paginator=paginator,
+                    ):
+                        if parent_record:
+                            for child_record in child_page:
+                                child_record.update(parent_record)
+                        yield child_page
+
+            resources[resource_name] = dlt.resource(
+                paginate_dependent_resource,
+                data_from=predecessor,
+                **resource_kwargs,
+            )(
+                method=endpoint_config.get("method", "get"),
+                path=endpoint_config.get("path"),
+                params=request_params,
+                paginator=paginator,
+            )
+
+    return list(resources.values())
+
+
+def make_endpoint_resource(
+    resource: Union[str, EndpointResource], default_config: EndpointResource
+):
+    """
+    Creates an EndpointResource object based on the provided resource
+    definition and merges it with the default configuration.
+
+    This function supports defining a resource in multiple formats:
+    - As a string: The string is interpreted as both the resource name
+        and its endpoint path.
+    - As a dictionary: The dictionary must include `name` and `endpoint`
+        keys. The `endpoint` can be a string representing the path,
+        or a dictionary for more complex configurations. If the `endpoint`
+        is missing the `path` key, the resource name is used as the `path`.
+    """
+    if isinstance(resource, str):
+        resource = {"name": resource, "endpoint": {"path": resource}}
+        return deep_merge(copy.deepcopy(default_config), resource)
```

Comment:
better to use deep_merge on empty dict to get an initial copy. we do not want to clone objects. especially if for example someone wants to authenticate authenticator before passing